src/expire.c

```diff
@@ -584,14 +588,31 @@ void ttlGenericCommand(client *c, int output_ms) {
     }
 }
 
-/* TTL key */
+/* TTL key [ABS] */
 void ttlCommand(client *c) {
-    ttlGenericCommand(c, 0);
+    ttlGenericCommand(c, 0, 0);
 }
 
-/* PTTL key */
+/* PTTL key [ABS] */
 void pttlCommand(client *c) {
-    ttlGenericCommand(c, 1);
+    ttlGenericCommand(c, 1, 0);
+}
+
+/* EXPIRETIME key TIME|PTIME */
+void expiretimeCommand(client *c) {
+    int output_ms = 0;
+
+    /* Only a third argument is allowed */
+    if (c->argc == 3 && !strcasecmp(c->argv[2]->ptr,"time")) {
+        output_ms = 0;
+    } else if (c->argc == 3 && !strcasecmp(c->argv[2]->ptr,"ptime")) {
+        output_ms = 1;
+    } else {
+        addReplySubcommandSyntaxError(c);
```

Comment:
This is a syntax error, not a subcommand syntax error.
```suggestion
        addReplyErrorObject(c,shared.syntaxerr);
```

Reply:
Done